contracts/GenericLender/GenericAave.sol

```diff
@@ -0,0 +1,176 @@
+ *
+ ********************* */
+
+contract GenericAave is GenericLenderBase {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+    
+    uint256 private constant blocksPerYear = 2_300_000;
+
+    IProtocolDataProvider public immutable protocolDataProvider;
+    IAToken public immutable aToken;
+
+    constructor(
+        address _strategy,
+        string memory name,
+        IProtocolDataProvider _protocolDataProvider,
+        IAToken _aToken
+    ) public GenericLenderBase(_strategy, name) {
+        protocolDataProvider = _protocolDataProvider;
+        aToken = _aToken;
+
+        require(ILendingPool(_protocolDataProvider.ADDRESSES_PROVIDER().getLendingPool()).getReserveData(address(want)).aTokenAddress == address(_aToken), "WRONG ATOKEN");
+
+        want.approve(address(_protocolDataProvider.ADDRESSES_PROVIDER().getLendingPool()), type(uint256).max);
+    }
+
+    function nav() external view override returns (uint256) {
+        return _nav();
+    }
+
+    function _nav() internal view returns (uint256) {
+        return want.balanceOf(address(this)).add(underlyingBalanceStored());
+    }
+
+    function underlyingBalanceStored() public view returns (uint256 balance) {
+        balance = aToken.balanceOf(address(this));
+    }
+
+    function apr() external view override returns (uint256) {
+        return _apr();
+    }
+
+    function _apr() internal view returns (uint256) {
+        return uint(_lendingPool().getReserveData(address(want)).currentLiquidityRate).div(1e9); // dividing by 1e9 to pass from ray to wad
+    }
+
+    function weightedApr() external view override returns (uint256) {
+        uint256 a = _apr();
+        return a.mul(_nav());
+    }
+
+    function withdraw(uint256 amount) external override management returns (uint256) {
+        return _withdraw(amount);
+    }
+
+    //emergency withdraw. sends balance plus amount to governance
+    function emergencyWithdraw(uint256 amount) external override management {
+        _lendingPool().withdraw(address(want), amount, address(this));
+
+        want.safeTransfer(vault.governance(), want.balanceOf(address(this)));
+    }
+
+    //withdraw an amount including any want balance
+    function _withdraw(uint256 amount) internal returns (uint256) {
+        uint256 balanceUnderlying = aToken.balanceOf(address(this));
+        uint256 looseBalance = want.balanceOf(address(this));
+        uint256 total = balanceUnderlying.add(looseBalance);
+
+        if (amount > total) {
+            //cant withdraw more than we own
+            amount = total;
+        }
+
+        if (looseBalance >= amount) {
+            want.safeTransfer(address(strategy), amount);
+            return amount;
+        }
+
+        //not state changing but OK because of previous call
+        uint256 liquidity = want.balanceOf(address(aToken));
+
+        if (liquidity > 1) {
```

Comment:
why `1`? comment plz.

Reply:
asked this to @Grandthrax a while ago. he wasn't sure about why using 1 instead of 0 but I kept this because GenericCream is using it and working (so far)